main.py

```diff
@@ -153,14 +169,72 @@ async def on_post(self, req: Request, resp: Response):
         resp.data = completion.to_json()
 
 
+class Embeddings:
+    def __init__(self, model_name: str) -> None:
+        self.model_name = model_name
+
+    async def on_post(self, req: Request, resp: Response):
+        buf = await req.stream.readall()
+        try:
+            # todo: llmspec hasn't implemented from_bytes for EmbeddingRequest
+            embedding_req = msgspec.json.decode(buf, type=EmbeddingRequest)
+        except msgspec.ValidationError as err:
+            logger.info(f"Failed to parse request: {err}")
+            resp.status = falcon.HTTP_400
+            resp.status = falcon.HTTP_400
+            resp.data = ErrorResponse.from_validation_err(err, str(buf)).to_json()
+            return
+
+        model = SentenceTransformer(self.model_name)
+        embeddings = model.encode(embedding_req.input)
+        # convert embeddings of type list[Tensor] | ndarray to list[float]
+        if isinstance(embeddings, list):
+            embeddings = [e.tolist() for e in embeddings]
+        elif isinstance(embeddings, torch.Tensor):
+            embeddings = embeddings.tolist()
+        else:
+            embeddings = embeddings.tolist()
+
+        embedding_data = EmbeddingData(embedding=embeddings, index=0)
+        embedding_resp = EmbeddingResponse(
+            data=embedding_data,
+            model=self.model_name,
+            usage=TokenUsage(
+                prompt_tokens=0,  # No prompt tokens, only embeddings generated.
+                completion_tokens=0,  # No completions performed, only embeddings generated.
+                total_tokens=len(embeddings),
+            ),
+        )
+        # todo: llmspec hasn't implemented to_json for EmbeddingResponse
+        resp.data = msgspec.json.encode(embedding_resp)
+
+
+class EmbeddingsEngineRouteWrapper(Embeddings):
+    def __init__(self, model_name: str) -> None:
+        super().__init__(model_name)
+
+    async def on_post(self, req: Request, resp: Response, engine: str):
+        await super().on_post(req, resp)
```

Comment:
Seems it equals to the parent class?

Reply:
parent class doesn't accept the engine arg. Do you want me to add the arg to Embeddings?